### ISSUE TITLE
Better assert default failure message, like that given by `@test`

### ISSUE THREAD

**Issue by oxinabox** [JuliaLang/julia#21725]
when you fail a `@test`, you get a great message telling you what the expect value was and what the actual value was.
With `@assert`, you do not.

For example
```julia
julia> using Base.Test

julia> a=(2,4)
(2,4)

julia> @test a[1]==3
Test Failed
  Expression: a[1] == 3
   Evaluated: 2 == 3
ERROR: There was an error during testing
 in record(::Base.Test.FallbackTestSet, ::Base.Test.Fail) at ./test.jl:397
 in do_test(::Base.Test.Returned, ::Expr) at ./test.jl:281

julia> @assert a[1]==3
ERROR: AssertionError: a[1] == 3

```

I often fine myself writing things like:
```julia
julia> @assert(a[1]==3, "a[1]==$(a[1]) != 3")
ERROR: AssertionError: a[1]==2 != 3
```
but `@test` shows that this can be done automatically.
I think that makes a better default, than just printing out what the expression was.

**Comment by TotalVerb** [comment]
Just curious: why not simply use `Test.@test`?

**Comment by oxinabox** [comment]
They are not semantically the same operation.
That feels rather dirty, but it would work, yes.

But  eventually `@assert`s will be (to some lesser or greater extent),
be removed when running in some form of optimized mode.
https://github.com/JuliaLang/julia/issues/7732
https://github.com/JuliaLang/julia/issues/10614

**Comment by StefanKarpinski** [comment]
I agree with this general direction – if we're going to keep asserts. There's the question of whether we should or not, however, since people tend to abuse asserts for error checking instead strictly using them for verifying the internal logic of their algorithms when testing.

**Comment by samoconnor** [comment]
Related #15495

**Comment by alexeypetrushin** [comment]
+1 for printing the actual value, currently you see message like `ERROR: LoadError: AssertionError: base ≈ 12` and you need to add `println(base)` and run it again before you can fix the code.

**Comment by Wikunia** [comment]
Maybe there can be a different macro for this behavior otherwise @StefanKarpinski or should one always do `using Test; @test` for this usecases?
